### ISSUE TITLE
bug: iOS 14 Black screen always on initial app launch prior to actually showing the splash screen [SplashScreen]

### ISSUE THREAD

**Issue by Exocomp** [ionic-team/capacitor#3589]
This bug is for the SplashScreen plugin described [here](https://capacitorjs.com/docs/apis/splash-screen).

Create an ionic app with capacitor, deploy it on the simulator or physical device:

1) Launch the application
2) Get black screen <-- this was not here prior to iOS 14
3) Transition to the actual splash image

Here is an example of a project where you can reproduce it: https://github.com/Exocomp/demo-ionic-swippable-modal.git

```
Ionic:

   Ionic CLI                     : 6.11.8 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.2.3
   @angular-devkit/build-angular : 0.1000.8
   @angular-devkit/schematics    : 10.0.8
   @angular/cli                  : 10.0.8
   @ionic/angular-toolkit        : 2.3.3

Capacitor:

   Capacitor CLI   : 2.4.1
   @capacitor/core : 2.4.1

Utility:

   cordova-res : 0.15.1
   native-run  : 1.1.0

System:

   NodeJS : v10.13.0 (/usr/local/bin/node)
   npm    : 6.4.1
   OS     : macOS Catalina
```

**Comment by AE1NS** [comment]
Same here. Does someone have a workaround/quickfix for this?

**Comment by Deepfreezed** [comment]
Launch screen with image stopped working with iOS 14, Xcode 12. Only workaround currently is to not use an image.

**Comment by media4learning** [comment]
Same issue here, ios 14 fresh cap app launch screen on ios is black. If anyone finds a solution please post

**Comment by bolivir** [comment]
I see the same issue happening, launch app --> black screen --> Splash

**Comment by francodelsancio** [comment]
I see the same issue happening, workaround for now, not use background image, replace it with a view and make the launch screen in there by hand.

**Comment by chrisdelambert** [comment]
I have noticed a lot of apps displaying a black screen briefly before the launch screen on ios 14. I changed my SplashScreen plugin config to the below and the launch screen now displays: (delete the delay param)

"SplashScreen": {
      "launchAutoHide": false
   }

**Comment by Exocomp** [comment]
> I see the same issue happening, workaround for now, not use background image, replace it with a view and make the launch screen in there by hand.

@francodelsancio can you please provide some steps on how to do this.

@chrisdelambert "launchAutoHide": false doesn't work still has the black screen.

**Comment by francodelsancio** [comment]
> @francodelsancio can you please provide some steps on how to do this.
Sure, i remove the ImageView in LaunchScreen.storyboard and set a view instead, then i set the background color to the view, and added the logo with auto layout. I noticed that the images appear in the SDK but not in the device unless they are added in Bundle Resources (Build Phases).

<img width="175" alt="Captura de Pantalla 2020-09-30 a la(s) 09 46 25" src="https://user-images.githubusercontent.com/18752001/94686862-d0ed5f00-0301-11eb-90bc-5acb4db910f8.png">

**Comment by Exocomp** [comment]
@francodelsancio thanks for the steps, I tried it and it's not solving the issue. Correct me if I'm wrong but your steps just set the initial logo it doesn't affect the splash screen.  The problem I see is that the transition to the splash screen is a black.

So this is what I see after your steps:

1) Launch app
2) See logo
3) Transition from logo to black screen
4) Transition from black screen to splash screen

So unfortunately it is still not working.

**Comment by bolivir** [comment]
> I have noticed a lot of apps displaying a black screen briefly before the launch screen on ios 14. I changed my SplashScreen plugin config to the below and the launch screen now displays: (delete the delay param)
> 
> "SplashScreen": {
> "launchAutoHide": false
> }

Above was the solution for me, i removed the other entries i had in there like launchShowDuration.
Now my config only contains the launchAutohide key. I just rebuilded the app and now i see no black screen appearing.

**Comment by francodelsancio** [comment]
@Exocomp i am sorry to hear that this not solve your problem, it seems, in my case, that the back screen is the background color coming from the imageView, because at run time its not loading from the resources, thats why i modify the launchscreen.storyboard, remove the imageView, added a view, set the background color, and put the logo. 
In configuration I also set it to start with that storyboard. Regards

**Comment by Exocomp** [comment]
@francodelsancio it makes sense what your saying but as soon as I add an image it goes to black, can you please include your LaunchScreen.storyboard file that will really help.

**Comment by francodelsancio** [comment]
@Exocomp i attach above an image of the tree in the splashscreen.storyboard if you change the background color of the elements you will notice that it is not loading the images. to load them you must attach them in the resource bundle. The assets folder are in the resource bundle, why are not loaded I don’t know. But the workaround works. All this assuming that we have the same problem. Regards

**Comment by JU1CYC0DY** [comment]
Are there any updates on this problem here? Facing the same issues with the example app and its a pain point for me to see every time this black screen. 
Appreciate every help/workaround; the ones mentioned above didn't work for me.

Best regards

**Comment by DevelopmentLocklear** [comment]
Same problem here, black screen of the dead... it's totally anoying for user experience. I'v not found workaround... any solution here ? an update ?

**Comment by Faisal49m** [comment]
Same issue here, tried all workarounds explained above didn't work, I was able to fix it in Cordova by removing the UILaunchStoryboardName from the .plist file but it didn't work here

**Comment by eddyD45** [comment]
I've tried all the proposed workarounds posted above and still have the same issue, usually with the black screen lasting for about 30 seconds to a minute before fully launching the app. It only happens on the first launch after installation; opening the app again a second time does not yield the black screen anymore. Are there any updates on this issue?

**Comment by kinggolf** [comment]
TLDR: What worked for me was to reduce file size of launch images.

I have 3 current Ionic/Capacitor/Angular apps. Two of them exhibit the same problem described here, initial black screen on app start prior to displaying actual launch image. The third never showed this problem and always started up fine.

After digging through all set-ups, configs, dependencies, ..., I found nothing to help my 2 problem apps. Just this morning I noticed the file size differences of the launch images:
good app: 29KB
problem apps: 145KB and 104KB

First I just pulled the launch images from the good app to one of problem apps and launch screen at start up was perfect, i.e. no black screen. Then I ran the launch images from 2 problem apps through [squoosh](https://squoosh.app/) - 145KB images reduced to 37KB and 104KB images reduced to 45KB. Put these new launch screen images into respective iOS apps and built onto my iPhone. Both app now start up perfectly with no black screen image.

Interesting note - both problematic apps are older and previously released to AppStore and have always worked fine. Something with iOS 14 definitely created this issue. Issue is not related to Xcode 12.

Possibly [cordova resources](https://ionicframework.com/docs/cli/commands/cordova-resources) would help this if used, but I have not tried.

I learned about squoosh from [React Native Radio Podcast](https://reactnativeradio.com/episodes/176-react-native-final-steps-with-gant-laborde-04Hu21U2) one of the most informative dev podcasts I've come across - even though I do no React Native development.

**Comment by nachshonfertel** [comment]
> I have 3 current Ionic/Capacitor/Angular apps. Two of them exhibit the same problem described here, initial black screen on app start prior to displaying actual launch image. The third never showed this problem and always started up fine.

Tried this and it does not work for me.

**Comment by kinggolf** [comment]
Additional set up info in my apps, in case it helps, I am not using any splash screen plugin (Capacitor or Cordova). And no splash screen settings in `capacitor.config.json`. Just seems to work better for me.

**Comment by eddyD45** [comment]
> TLDR: What worked for me was to reduce file size of launch images.
> 
> I have 3 current Ionic/Capacitor/Angular apps. Two of them exhibit the same problem described here, initial black screen on app start prior to displaying actual launch image. The third never showed this problem and always started up fine.
> 
> After digging through all set-ups, configs, dependencies, ..., I found nothing to help my 2 problem apps. Just this morning I noticed the file size differences of the launch images:
> good app: 29KB
> problem apps: 145KB and 104KB
> 
> First I just pulled the launch images from the good app to one of problem apps and launch screen at start up was perfect, i.e. no black screen. Then I ran the launch images from 2 problem apps through [squoosh](https://squoosh.app/) - 145KB images reduced to 37KB and 104KB images reduced to 45KB. Put these new launch screen images into respective iOS apps and built onto my iPhone. Both app now start up perfectly with no black screen image.
> 
> Interesting note - both problematic apps are older and previously released to AppStore and have always worked fine. Something with iOS 14 definitely created this issue. Issue is not related to Xcode 12.
> 
> Possibly [cordova resources](https://ionicframework.com/docs/cli/commands/cordova-resources) would help this if used, but I have not tried.
> 
> I learned about squoosh from [React Native Radio Podcast](https://reactnativeradio.com/episodes/176-react-native-final-steps-with-gant-laborde-04Hu21U2) one of the most informative dev podcasts I've come across - even though I do no React Native development.

Strangely enough, this _somewhat_ fixed the issue for me. After reducing the splash image size, my app will load the splash image immediately. However, this only seems to be masking the underlying issue, as I still have a 1~2 minute hang-time before my app properly loads and hides the splash image. Either way, this fixes the original issue at hand, so thank you.

**Comment by arcbus** [comment]
I'll just throw my hat in the ring and also mention that i'm seeing this.

Had an older cordova project where indeed removing the UILaunchStoryboardName from plist worked, but haven't been able to solve it in capacitor :(

**Comment by aparajita** [comment]
Hang in there, next week I'll be releasing a new splash screen plugin that will fix all of these problems.

**Comment by Exocomp** [comment]
It turned out to be an image resolution for me, it was 2700x2700 initially, I dropped it down to 2300x2300 and I don't get the black screen anymore. If you try this just make sure to backup your original image so once this issue is fixed you can start using the higher resolution again. Note that the size on the disk stayed pretty much the same so not a size on disk issue. Thumbs up if this works for you too. 👍

**Comment by yacut** [comment]
@aparajita Any progress on this? We really waiting for this fix... 😞

**Comment by aparajita** [comment]
@yacut Hopefully this week. I'm actually working on 3 killer plugins: splash screen, biometric auth, and console. It will be worth the wait. 😁

**Comment by angelolencina** [comment]
@aparajita hi my friend! any progress on this?

**Comment by aparajita** [comment]
@angelolencina 3 killer plugins became 4 😁, and I had a lot of work to do for my job, so it was a busy week. It looks good for releasing the splash screen plugin this week.

**Comment by angelolencina** [comment]
> @angelolencina 3 killer plugins became 4 😁, and I had a lot of work to do for my job, so it was a busy week. It looks good for releasing the splash screen plugin this week.

Ok. Thank you!!  😁👍

**Comment by AE1NS** [comment]
Hey @aparajita, do you have any news about the splash screen plugin?

At the moment our iOS App has a black screen and the Android App has a colored background (instead of a stretched image) before the ionic splash is displayed. Unfortunately there seems to be no proper solution until you rescue us 👍 Thank you for your effort, this sound really great!

**Comment by aparajita** [comment]
This week for sure. Just releasing a secure storage plugin today, splash screen will be next.

**Comment by bradmartin** [comment]
@aparajita - I'm coming here from searching on solutions for other frameworks. This issue is with iOS 14. I've been trying to implement the `UILaunchScreen` key in the plist of project by setting the `UIImageName` to an imageasset declared in the assets. However, the sizing is very difficult to determine for all the iPhone screens. 

Curious how you're approaching this. As this key is also something only for iOS 14, and if you support previous iOS versions the storyboard is supposed to be working. I have a worry this is something we're going to be waiting on an iOS patch release for this to stabilize.

Would appreciate any insight you have regarding this, as I'm digging deep into native projects and still can't find a solid solution. See: https://stackoverflow.com/questions/63978396/launch-screen-not-working-on-ios-14-with-xcode-12 for more context on the impact hitting developers not just using Capacitor or Ionic as it's iOS 14 that seems to have the issue respecting the previous storyboard configuration. Cheers :)

**Comment by aparajita** [comment]
@bradmartin I wouldn't spend too much time on it, I will be releasing my Super Splash Screen plugin tomorrow that does everything you want.

**Comment by bradmartin** [comment]
I don't think any plugin in any framework is going to fix the issue as this seems to be something in iOS 14 causing it.

Storyboards for the launch screen work according to this table below:

```iPad (8th generation) - SUCCESS
iPad Air (4th generation) - SUCCESS
iPad Pro (9.7 inch) - SUCCESS
iPad Pro (11 inch  2nd generation) - SUCCESS
iPad Pro (12.9 inch 4th generation) - SUCCESS

iPhone 8 - SUCCESS
iPhone 8 Plus - FAILURE
iPhone 11 - SUCCESS
iPhone 11 Pro - FAILURE
iPhone 11 Pro Max - FAILURE
iPhone 12 - FAILURE
iPhone 12 Pro - FAILURE
iPhone 12 Pro Max - FAILURE
iPhone 12 mini - FAILURE

iPhone SE (2nd generation) - SUCCESS
iPod Touch (7th generation) - SUCCESS
```

So it seems that the storyboard for launch screen is failing on "special" devices like the plus, pro max, etc. 

According to apple, storyboards should work on iOS 14, since if you want to support anything prior to 14 you cannot use the new `UILaunchScreen` as part of the plist file to specify a `UIImage` to render.

**Comment by franciscomontiel** [comment]
I am having the same problem and still decided to push to testflight/appstore , the problem with the black screen that happened in our development cellphones did not happened when downloading from store. Hope it helps.

**Comment by aparajita** [comment]
@bradmartin Storyboards are working for me on iOS 14 on those devices, at least in the simulator.

**Comment by bolivir** [comment]
@aparajita is the update live? Or when do you expect it to go live?

**Comment by bradmartin** [comment]
@aparajita - I've actually had a colleague have his storyboards working as well. It has to be a combination of iOS, devices and possibly xcode 12. Too many reports of the black launch screen on iOS with storyboards for it to not be an "issue" somewhere. I'm glad some people aren't encountering it, but it's a bummer because a lot of people are still facing it across various frameworks and apps.

See [this SO post](https://stackoverflow.com/questions/63978396/launch-screen-not-working-on-ios-14-with-xcode-12) to get an idea of how spread the issue is and the various approaches devs have taken to resolve, but with no one having concrete information on a solution. Again, that's great you're not facing the issue, but sadly, it's an issue somewhere and I've not found any solid information to address it anywhere. Since the documentation states that it should work just fine :)

**Comment by aparajita** [comment]
@bradmartin That's an iOS bug that has nothing to do with Capacitor, so technically the discussion doesn't belong here.

@bolivir Hopefully going live today, have to write documentation.

**Comment by bradmartin** [comment]
May not belong here, but I'm inquiring because I wonder if you're going to run into the same issues with your plugin in Capacitor, at any rate best of luck but looks like there is no solid solution even if you have a working plugin since this is an iOS issue impacting native iOS apps and other frameworks that build native apps. Cheers.

**Comment by calvinwoo** [comment]
What ended up solving the problem for me was following the instructions in this comment: https://github.com/apache/cordova-ios/issues/990#issuecomment-705086806

I removed `UILaunchStoryboardName` from `Info.plist`. Then, I added a dictionary called `UILaunchScreen` and within that, a key `UIImageName` set to the value `LaunchScreen`.

**Comment by mugenyi** [comment]
<key>UILaunchScreen</key>
      <dict>
        <key>UIImageName</key>
        <string>LaunchScreen</string>
      </dict>

**Comment by Syahrul** [comment]
@erikmay in ios 14.6, it always show a white screen instead of the color that you set (actual device). How can I fix this?
Thank you

**Comment by rgbaman** [comment]
> ### After having dealt with the problem extensively, I am now happy to probably be able to provide all of you with **a solution to the problem.**
> **1.** Search for and remove this following part from the `Info.plist`:
> 
> `<key>UILaunchStoryboardName</key> <string>LaunchScreen</string>`
> 
> **2.** Instead add this part into the `Info.plist`:
> 
> `<key>UILaunchScreen</key> <dict> <key>UIColorName</key> <string>Splashscreen-Color</string> <key>UIImageRespectsSafeAreaInsets</key> <true/> </dict>`
> 
> **3.** Now we can change the black screen at the start of the app to any color we want / need. This is much better user experience wise. Add your colors in the `Assets.xcassets`:
> 
> <img alt="Bildschirmfoto 2021-03-09 um 22 40 39" width="1127" src="https://user-images.githubusercontent.com/36517827/110541789-8251b580-8128-11eb-9041-278c7a27372b.png">
> 
> As you can see I called my colors in the `Assets.xcassets`: `"Splashscreen-Color"`
> This name is also mentioned in our `Info.plist`. Right here -> `<string>Splashscreen-Color</string>`
> If you want to change the name, change it in the `Assets.xcassets` and also the `Info.plist`.
> 
> This should be all steps necessary to remove the annoying black splash screen before our own splash screen.
> 
> **4.** Additionally you can also fully disable our own splash screen made by Capacitorjs and only use this color as a splash screen.
> To do that just add this to your `capacitor.config.json`:
> 
> `"SplashScreen": { "launchShowDuration": 0 }`
> 
> And you can also add your own image to the `Info.plist`-splash screen. Just add your image file with _a lot of transparency_ around the actual logo to the `Assets.xcassets`:
> 
> <img alt="Bildschirmfoto 2021-03-09 um 22 50 48" width="1127" src="https://user-images.githubusercontent.com/36517827/110542990-0b1d2100-812a-11eb-8355-80238443744a.png">
> 
> And then just add the following to your `Info.plist`:
> 
> `<key>UILaunchScreen</key> <dict> <key>UIColorName</key> <string>Splashscreen-Color</string> <key>UIImageName</key> <string>Splashscreen-Image</string> <key>UIImageRespectsSafeAreaInsets</key> <true/> </dict>`
> 
> ### I hope this solves the problem for everyone! Please tell me if it worked for you.

This no longer works for iOS 14.6 and it displays a white screen before navigating to the first page.

**Comment by haroonakram** [comment]
Does anyone find any fix for this problem... Facing the same problem and Apple rejected the build due to this 

@erikmay tried your solution but not working

**Comment by eliassparagis** [comment]
I had the same issue until I replaced the splash png files with smaller ones. After that black screen gone and the capacitor livereload works as expected.
Working on mac M1 Big Sur 11.4

**Comment by hbjpn** [comment]
Reducing the image size(in pixel) of the 3 files in the following folder works for me ( iOS 14.6 and Xcode 12.5 with ionic 5 and capacitor 3.0)
Assets.xcassets/Splash.imageset
splash-2732x2732.png
splash-2732x2732-1.png
splash-2732x2732-2.png

I resized the images from 2732x2732 (image size generated by cordova-res) to 2048x2048. Maybe reducing the image size less than 2732 pixel will result in slightly blurred splash image in e.g. iPad Pro 12 inch, but anyway splash screen works.

Note that the image "file" size seems to be not a matter as I confirmed 2732x2732 with 140KB does not work but 2048x2048 with 200KB images works. 

This seems to be a problem in iOS 14, Apple should fix this issue. 2732 pixel splash screen is definitely needed to cater for the latest devices.

**Comment by dkiselew** [comment]
I solved this problem by adding in XCode "iPhone" images to Splash.imageset with corresponding sizes (see [Apple docs](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/#device-screen-sizes-and-orientations)), looks like iOS 14 doesn't apply Universal image correctly:

<img width="922" alt="Screenshot 2021-09-08 at 11 40 13" src="https://user-images.githubusercontent.com/1822893/132477251-3e489cd2-5301-4ad7-96f1-fb11b9e33fd7.png">

More about this: https://github.com/apache/cordova-ios/issues/990

**Comment by franciscogar94** [comment]
> I solved this problem by adding in XCode "iPhone" images to Splash.imageset with corresponding sizes (see [Apple docs](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/#device-screen-sizes-and-orientations)), looks like iOS 14 doesn't apply Universal image correctly:
> 
> <img alt="Screenshot 2021-09-08 at 11 40 13" width="922" src="https://user-images.githubusercontent.com/1822893/132477251-3e489cd2-5301-4ad7-96f1-fb11b9e33fd7.png">
> 
> More about this: [apache/cordova-ios#990](https://github.com/apache/cordova-ios/issues/990)

@dkiselew 
Hi. What was the dimensions that u use for the Iphone Splash part?

**Comment by EbenDang** [comment]
> > I solved this problem by adding in XCode "iPhone" images to Splash.imageset with corresponding sizes (see [Apple docs](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/#device-screen-sizes-and-orientations)), looks like iOS 14 doesn't apply Universal image correctly:
> > <img alt="Screenshot 2021-09-08 at 11 40 13" width="922" src="https://user-images.githubusercontent.com/1822893/132477251-3e489cd2-5301-4ad7-96f1-fb11b9e33fd7.png">
> > More about this: [apache/cordova-ios#990](https://github.com/apache/cordova-ios/issues/990)
> 
> @dkiselew Hi. What was the dimensions that u use for the Iphone Splash part?

I use 375 * 812 (iPhoneX@1x)

**Comment by dkiselew** [comment]
> > I solved this problem by adding in XCode "iPhone" images to Splash.imageset with corresponding sizes (see [Apple docs](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/#device-screen-sizes-and-orientations)), looks like iOS 14 doesn't apply Universal image correctly:
> > <img alt="Screenshot 2021-09-08 at 11 40 13" width="922" src="https://user-images.githubusercontent.com/1822893/132477251-3e489cd2-5301-4ad7-96f1-fb11b9e33fd7.png">
> > More about this: [apache/cordova-ios#990](https://github.com/apache/cordova-ios/issues/990)
> 
> @dkiselew Hi. What was the dimensions that u use for the Iphone Splash part?

2x image is iPhone 7/8 resolution: 750x1334 px
3x image is iPhone X resolution: 1125x2436 px
For 1x image you can use 2x image

**Comment by franciscogar94** [comment]
@dkiselew  Thanks.
@EbenDang Thanks 
I manage to resolve the problem using this 1125x2436 px resolution.

**Comment by daniel-7235** [comment]
Reducing the image resolution no longer helps for iOS 15.

**Comment by daviesdoclc** [comment]
For me the issue seems to be cordova-res.  It only generates 

splash-2732x2732.png
splash-2732x2732-1.png
splash-2732x2732-2.png

It also needs to generate a splash-1334x1334.png and then add the additional references in the Content.json appropriately

    {
      "idiom" : "iphone",
      "filename" : "splash-1334x1334.png",
      "scale" : "1x"
    },
    {
      "idiom" : "iphone",
      "filename" : "splash-1334x1334.png",
      "scale" : "2x"
    },
    {
      "idiom" : "iphone",
      "filename" : "splash-1334x1334.png",
      "scale" : "3x"
    },

I guess I'd also argue that you don't need 3 splash-2732x2732.png files.  Just the one for all 3 scales would work?

**Comment by Syahrul** [comment]
None of these workaround work in ios 15.
I'm using Ionic 6, capacitor 3 and the black background still appears no matter what size or image I'm using.

One thing I noticed playing with Xcode. In file LaunchScreen.storyboard > View Controller Scene > View Controller > Splash
If I change the Image to one of the system icon?, the black background disappear and the splash screen work just fine. It display the splash screen with chosen icon as background. Maybe some one can shed some light on how to fix this?

![Screenshot 2022-03-05 at 4 05 20 AM](https://user-images.githubusercontent.com/2231998/156834149-49d9579e-43a4-4d0b-abb6-30c8f5c9cc67.png)

**Comment by PeiJueChen** [comment]
> 

Thx. This works for me. Remember to del app, restart the phone and install it again, or try to install it on another phone to see if it works. Because apple will cache the splash, it is difficult to test whether it works or not.

**Comment by Lzolcsi** [comment]
@Syahrul I can confirm this behaviour with the system icons.
Unfortunately none of the solutions mentioned above worked for me.

**Comment by TheAngularGuy** [comment]
None of the solutions above worked for me. Still looking for a way to solve this

**Comment by jblackabraham** [comment]
Hi guys, really resizing the size will make it works, i just changed the splash screen to a smaller image make it works,
here's the reference explaining it in a better way why => https://stackoverflow.com/questions/63978396/launch-screen-not-working-on-ios-14-with-xcode-12

**Comment by giovw** [comment]
My solution was to change the mode image in Photoshop to Indexed color.
<img width="360" alt="Screen Shot 2022-07-12 at 00 28 31" src="https://user-images.githubusercontent.com/2634691/178415736-a7c7b65f-e9d8-447f-9101-59b1ee94454a.png">
Open in Photoshop
<img width="637" alt="Screen Shot 2022-07-12 at 00 27 54" src="https://user-images.githubusercontent.com/2634691/178415842-7edc898a-0bb2-4749-bac6-8e3e99e057f4.png">
It worked for me.
<img width="755" alt="Screen Shot 2022-07-12 at 00 32 47" src="https://user-images.githubusercontent.com/2634691/178416335-4b510347-2ff7-4590-94ad-1884747f5929.png">

**Comment by leixdd** [comment]
I fixed this issue in my project using this answer from stackoverflow.

https://stackoverflow.com/questions/58094614/ionic-capacitor-black-screen-after-splash

**Comment by aparajita** [comment]
Hey folks, if you're looking for a better splash screen solution, [@aparajita/capacitor-splash-screen](https://github.com/aparajita/capacitor-splash-screen) v2 has just been released. Check it out!

**Comment by weihsth** [comment]
For me an entry was missing in the Info:

Key: Main storyboard file base name
Value: Main

**Comment by elmartino** [comment]
What helped for me to get rid of the black screen was:

1. In xcode open LaunchScreen
2. Select splash under the view controller
3. At the top right set 'Highlighted' also to splash

**Comment by mendo96** [comment]
> In my case, the splash image was not shown, just see white blank image. ( iOS 14.6 and Xcode 12.5 with ionic 5 and capacitor 3.0)
> 
> Reducing the image size(in pixel) of the 3 files in the following folder works for me. Assets.xcassets/Splash.imageset splash-2732x2732.png splash-2732x2732-1.png splash-2732x2732-2.png
> 
> I resized the images from 2732x2732 (image size generated by cordova-res) to 2048x2048. Maybe reducing the image size less than 2732 pixel will result in slightly blurred splash image in e.g. iPad Pro 12 inch, but anyway splash screen works.
> 
> Note that the image "file" size seems to be not a matter as I confirmed 2732x2732 with 140KB does not work but 2048x2048 with 200KB images works.
> 
> This seems to be a problem in iOS 14, Apple should fix this issue. 2732 pixel splash screen is definitely needed to cater for the latest devices.

Doing this worked perfectly on ionic 6 capacitor 4. Thank you !

**Important**: Delete the app, restart the device, install new.

**Comment by ipehimanshu** [comment]
Hello 

Does anyone have fix, 

we are facing this issue with capacitor 5.

Please help 

Thank you

**Comment by vochitan90** [comment]
for me the light mode alway show when launch screen.

I'm using capacitor 4

**Comment by TheAngularGuy** [comment]
Resizing the splashScreen Image to 2400x2400 (+ deleting the app and restarting the phone) solved the issue for me.

**Comment by joelrose2** [comment]
After reading tons of threads online and spending many hours trying to figure this out, I found a solution for eliminating the black screen that appears between the launch screen and the main app screen (something that I didn't see anyone else mention).

It is very simple: leave out the **.storyboard** extension of the name of the launch screen. With this small (and illogical) change, my launch screen remains visible until it is replaced by the actual app, no more black in between.

To solve this problem, I made this change in Info.plist:

From:
```
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen.storyboard</string>
```
To:
```
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
```

You can also make this change by editing the "Launch screen interface file base name" property value to remove the .storyboard part of the name shown there.

I hope this helps others!

**Comment by anthonyjuarezsolis** [comment]
> After reading tons of threads online and spending many hours trying to figure this out, I found a solution for eliminating the black screen that appears between the launch screen and the main app screen (something that I didn't see anyone else mention).
> 
> It is very simple: leave out the **.storyboard** extension of the name of the launch screen. With this small (and illogical) change, my launch screen remains visible until it is replaced by the actual app, no more black in between.
> 
> To solve this problem, I made this change in Info.plist:
> 
> From:
> 
> ```
>     <key>UILaunchStoryboardName</key>
>     <string>LaunchScreen.storyboard</string>
> ```
> 
> To:
> 
> ```
>     <key>UILaunchStoryboardName</key>
>     <string>LaunchScreen</string>
> ```
> 
> You can also make this change by editing the "Launch screen interface file base name" property value to remove the .storyboard part of the name shown there.
> 
> I hope this helps others!

Actually I've had it like this for 7 months and the same problem continues.
<img width="568" alt="image" src="https://github.com/ionic-team/capacitor/assets/38568593/0b7766d4-ae66-459b-980a-dacb711c9bf9">

**Comment by danielehrhardt** [comment]
Still no Capacitor fix here?

**Comment by sergiomilici** [comment]
Hello, I've tried many things, but none of them worked.
This worked for me:

* Resize all the splash images to be 2400x2400
* In `Info.plist` change `UILaunchStoryboardName` to 
```
<key>UILaunchStoryboardName</key>
<string>LaunchScreen.storyboard</string>
```
* Go to `App` -> `General` -> `App Icons and Launch Screen` and select `LaunchScreen.storyboard` and check `Include all app icons` (for this check I'm not sure the impact in the solution)
* Delete the app, restart the phone and install it again. (this was needed for updating the icons, but for SplashScreen I think it's not necessary since I did other changes and I saw the results without restating)

**NOTE**
After doing the previous steps, I noticed that the SplashScreen appeared instantly, but disappeared after a few seconds. It turned out that `launchShowDuration: 0` is not working correctly when `launchAutoHide: false`. As I understand, `launchShowDuration` is only used when launchAutoHide is false but this wasn't the case. Since I'm hiding it manually, I set 10 seconds of duration and now it's all working as expected.

**Project type:**

* New & clean Capacitor 5
* Added icons and splash screen  with `@capacitor/assets` as it is in the documentation.

**Comment by selected-pixel-jameson** [comment]
How has no one from the capacitor team weighed in on this? This issue has been open for 4 YEARS!!!

**Comment by nitzanrh** [comment]
Same here, using Ionic 7 with Capacitor 5 and none of the solutions above helped...

**Comment by joebnb** [comment]
> How has no one from the capacitor team weighed in on this? This issue has been open for 4 YEARS!!!

this is capacitor team,you join a real one，look this url path ，what‘s disappointed were they

https://github.com/ionic-team/capacitor/issues/7508

https://forum.ionicframework.com/t/its-a-nightmare-to-maintain-an-ionic-app/184652

https://mhamri.com/why-i-regret-using-ionic-for-app-development-c8b21b88d83a

**Comment by barart** [comment]
update?

**Comment by jacobg** [comment]
Having this issue in 2026 on Capacitor 8. Just migrated from Cordova, and Cordova did *not* have this issue. The suggestions in earlier comments did not resolve the issue.

**Comment by KAZAHv2** [comment]
Try making the splash screen image lighter, +- 100kb. To test, reinstall the apk and reboot the device before building.

**Comment by felixcarmona** [comment]
For anyone still hitting this in 2026: resizing the splash images + delete + restart + reinstall did not help me. The black appeared during the icon open animation and for ~2s before the real splash.

The key insight: the LaunchScreen.storyboard itself is fine. @capacitor/splash-screen instantiates that same storyboard at runtime and shows it correctly (I saw my splash right after the black). iOS just fails to render the storyboard during the system launch phase and paints black instead.

So instead of fighting the storyboard, I switched the launch screen to a solid color via the modern UILaunchScreen key, which iOS renders reliably.

1. Add a color set ios/App/App/Assets.xcassets/SplashBackground.colorset/Contents.json (use your
  background color; mine is #eef1f7):

```
  {
    "colors" : [
      {
        "color" : {
          "color-space" : "srgb",
          "components" : { "alpha" : "1.000", "red" : "0xEE", "green" : "0xF1", "blue" : "0xF7" }
        },
        "idiom" : "universal"
      }
    ],
    "info" : { "author" : "xcode", "version" : 1 }
  }
```

2. In Info.plist, replace UILaunchStoryboardName:

```
  <key>UILaunchStoryboardName</key>
  <string>LaunchScreen</string>
```

  with UILaunchScreen:

```
  <key>UILaunchScreen</key>
  <dict>
      <key>UIColorName</key>
      <string>SplashBackground</string>
  </dict>
```

3. Delete the app, restart the device (iOS caches the launch screen at the system level), then reinstall.

@capacitor/splash-screen keeps working: it falls back to the "LaunchScreen" storyboard by default (SplashScreen.swift#L91), so it still shows your real splash image after launch. Result: solid color during the open animation → plugin splash (logo) → app. No black frames.

If you want the logo in the OS launch screen too, add <key>UIImageName</key><string>Splash</string> to the dict,  but a plain color is the bulletproof part that kills the black.
